mozci/data/sources/treeherder/__init__.py

```diff
@@ -3,24 +3,69 @@
 from collections import defaultdict
 from functools import lru_cache
 
+from adr.util.memoize import memoized_property
 from loguru import logger
 
 from mozci.data.base import DataSource
 from mozci.errors import ContractNotFilled
+from mozci.util.req import get_session
 
 try:
     from treeherder.model.models import Job
 except ImportError:
     Job = None
 
 
-class TreeherderSource(DataSource):
-    """Uses ORM to query Treeherder."""
+class BaseTreeherderSource(DataSource):
+    pass
 
-    name = "treeherder"
+
+class TreeherderClientSource(BaseTreeherderSource):
+    """Uses the public API to query Treeherder."""
+
+    name = "treeherder_client"
+    supported_contracts = ("push_tasks_classifications",)
+    base_url = "https://treeherder.mozilla.org/api"
+
+    @memoized_property
+    def session(self):
+        session = get_session()
+        session.headers = {"User-Agent": "mozci"}
+        return session
+
+    def _run_query(self, query):
+        query = query.strip("/")
+        url = f"{self.base_url}/{query}"
```

Comment:
Nit: you could use urljoin

Reply:
Hm, `urljoin` seems to assume that there's a base component, so e.g:

```
urljoin("foo/bar", "baz") => "foo/baz"
urljoin("foo/", "baz") => "foo//baz"
```

Think I'll just leave this one as is.